tool/templates/dart/opaque.dart.jinja

```diff
@@ -4,20 +4,20 @@
 final class {{type_name}} implements ffi.Finalizable {
   final ffi.Pointer<ffi.Opaque> _underlying;
 
-  final core.List<Object> _edge_self;
-
-  {%- for lifetime in lifetimes.all_lifetimes() %}
-  final core.List<Object> _edge_{{lifetimes.fmt_lifetime(lifetime)}};
+  final core.List<Object> _edgeSelf;
+  {%- for lifetime in lifetimes %}
+  // ignore: unused element
+  final core.List<Object> _edge{{lifetime}};
   {%- endfor %}
 
   // Internal constructor from FFI.
-  // isOwned is whether this is owned (has finalizer) or not
-  // This also takes in a list of lifetime edges (including for &self borrows)
+  // This takes in a list of lifetime edges (including for &self borrows)
   // corresponding to data this may borrow from. These should be flat arrays containing
   // references to objects, and this object will hold on to them to keep them alive and
   // maintain borrow validity.
-  {{type_name}}._(this._underlying, bool isOwned, this._edge_self {%- for lifetime in lifetimes.all_lifetimes() -%} , this._edge_{{lifetimes.fmt_lifetime(lifetime)}} {%- endfor -%}) {
-    if (isOwned) {
+  {{type_name}}._(this._underlying, {core.List<Object> edgeSelf = const []{%- for lifetime in lifetimes -%}, required core.List<Object> edge{{lifetime}} {%- endfor -%}}) : this._edgeSelf = edgeSelf{%- for lifetime in lifetimes -%}, this._edge{{lifetime}} = edge{{lifetime}} {%- endfor %} {
+    if (this._edgeSelf.isEmpty) {
```

Comment:
issue: this isn't quite correct; a `'static` reference will be borrowed but have no edges.

My plan for this code was to have it null-check instead.

Reply:
https://github.com/rust-diplomat/diplomat/pull/413/files/37374e521db41a5faba0c941e1298a65c8874b4a#diff-40ee167005ede91c6f0e5aa1ca9d6c7666bb595acf34d61ea1a040044baf5537R892